app/src/main/java/com/example/sharewhatyoucanproject/deliverfood/DeliverViewModel.kt

```diff
@@ -0,0 +1,117 @@
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.FieldValue
+import com.google.firebase.firestore.FirebaseFirestore
+import java.util.*
+
+class DeliverViewModel(
+    private val db: FirebaseFirestore,
+) : ViewModel() {
+
+    private val _deliverResult = MutableLiveData<DeliverResult>()
+    val deliverResult: MutableLiveData<DeliverResult> = _deliverResult
+
+    private val _donationResult = MutableLiveData<DonationResult>()
+    val donationResult: MutableLiveData<DonationResult> = _donationResult
+
+    fun getItems() {
+        db.collection("posts")
+            .whereEqualTo("uid", FirebaseAuth.getInstance().currentUser!!.uid)
+            .whereNotEqualTo("status", 2)
+            .get()
+            .addOnCompleteListener { task ->
+                if (task.isSuccessful) {
+                    val list = mutableListOf<PostModel>()
+                    for (ds in task.result) {
+                        val postModel = ds.getGeoPoint("location")?.let {
+                            PostModel(
+                                ds.getString("imageUrl")!!,
+                                "" + ds.getString("title"),
+                                "" + ds.getString("description"),
+                                ds.getString("uid")!!,
+                                ds.getString("name")!!,
+                                ("" + ds["status"]).toInt(),
+                                ds.id,
+                                GeoPoint(it.latitude, it.longitude),
+                            )
+                        }
+
+                        if (postModel != null) {
+                            list.add(postModel)
+                        }
+                    }
+                    _deliverResult.value = DeliverResult.DeliveredSuccess(list)
+                } else {
+                    _deliverResult.value = DeliverResult.Error
+                }
+            }
+    }
+
+    fun donate(postModel: PostModel, requestModel: RequestModel) {
+        val donationmap: MutableMap<String, Any?> = HashMap()
+        donationmap["doner"] = FirebaseAuth.getInstance().currentUser!!.displayName
+        donationmap["donatedTo"] = requestModel.uid + ""
+        donationmap["receiverName"] = requestModel.name + ""
+        donationmap["time"] = FieldValue.serverTimestamp()
+        donationmap["title"] = postModel.title
+        donationmap["img"] = postModel.image
+        donationmap["desc"] = postModel.desc
+        donationmap["uid"] = FirebaseAuth.getInstance().currentUser!!.uid
+
+        val sref = db.collection("donations").document(randomRoom())
+        val uref =
+            db.collection("posts").document(postModel.postid)
+
+        val uref2 = db.collection("requests").document(requestModel.updateid)
+        db.runBatch { batch ->
+            batch.set(sref, donationmap)
+            batch.update(uref, "status", 2)
+            batch.update(uref2, "status", 2)
```

Comment:
What is 2 referring to? Could this be a const value or something else that would make it more clear?

Reply:
done